packages/bitcore-node/src/routes/api/address.ts

```diff
@@ -30,6 +31,66 @@ router.get('/:address', function(req, res) {
   ChainStateProvider.streamAddressUtxos(payload);
 });
 
+router.get('/:address/coins', async function (req, res) {
+  let { address, chain, network } = req.params;
+  // let { unspent, limit = 10, since } = req.query;
+  try {
+    let coins = await CoinStorage.collection.find({address, chain, network}).toArray();
+
+    const mintTxids : string[] = [];
+    const spentTxids : string[] = [];
+    const mintTxidIndexes = {};
+    const spentTxidIndexes = {};
+
+    coins.forEach((coin) => {
+      if(coin.mintTxid) {
+        mintTxids.push(coin.mintTxid);
+      } 
+      
+      if(coin.spentTxid) {
+        spentTxids.push(coin.spentTxid);
+      }
+
+    });
+
+    const mintTxidCoinPromises = mintTxids.map(function (txid, index) {
+      mintTxidIndexes[index] = txid;
+      return ChainStateProvider.getCoinsForTx({ chain, network, txid });
+    });
+
+    const spentTxidCoinPromises = spentTxids.map(function (txid, index) {
+      spentTxidIndexes[index] = txid;
+      return ChainStateProvider.getCoinsForTx({ chain, network, txid });
```

Comment:
You'll have to change this to 
```
CoinStorage.collection.find({chain, network, mintTxid: {$in: coins.map(c => c.spentTxid)}})
```

Otherwise you'll end up doing too many queries